hera_pspec/uvpspec.py

```diff
@@ -1382,6 +1390,33 @@ def read_from_group(self, grp, just_meta=False, spws=None, bls=None,
             self.cosmo = conversions.Cosmo_Conversions(
                                                 **ast.literal_eval(self.cosmo) )
 
+
+        # BACKWARDS COMPATIBILITY for files with Nblpairts.
+        # Check whether we are reading a pre Nbltpairs file by looking for the
+        # Nblpairts field.
+        if hasattr(self, "Nblpairts"):
+            reading_old_version = True
+            setattr(self, "Nbltpairs", self.Nblpairts)
+        else:
+            reading_old_version = False
+
+        if reading_old_version:
+            # If we are reading a power spectrum object that was written in the pre-Nbltpairs days
+            # then make sure to correctly set Ntimes and Ntpairs.
+            self.Ntimes = len(np.unique(np.hstack([self.time_1_array, self.time_2_array])))
+            self.Ntpairs = len(set([(t1, t2) for t1, t2 in zip(self.time_1_array, self.time_2_array)]))
+
+        # Clear deprecated attributes.
+        for dattr in self._meta_deprecated:
+            if hasattr(self, dattr):
+                to_delete = getattr(self, dattr)
+                del to_delete
+
+        for dattr in self._meta_dsets_deprecated:
+            if hasattr(self, dattr):
```

Comment:
Not covered by tests according to codecov

Reply:
See above.

Reply:
del to_delete won't successfully remove the attribute from the object. I change it to delattr.